Doc language edits - Part 1 (#732)
zama-ai/concrete-ml: docs/built-in-models/linear.md

|         [ElasticNet](../references/api/concrete.ml.sklearn.linear_model.md#class-elasticnet)         |             [ElasticNet](https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.ElasticNet.html#sklearn.linear_model.ElasticNet)             |
|       [SGDRegressor](../references/api/concrete.ml.sklearn.linear_model.md#class-sgdregressor)       |                           [SGDRegressor](https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.SGDRegressor.html)                           |

## Quantization parameters

The `n_bits` parameter controls the bit width of the inputs and weights of the linear models. Linear models do not use table lookups and thus alllows weight and inputs to be high precision integers.

> For models with input dimensions up to `300`, the parameter `n_bits` can be set to `8` or more. When the input dimensions are larger, `n_bits` must be reduced to `6-7`. Quantized models can preserve all performance metrics compared to the non-quantized float models from scikit-learn when `n_bits` is down to `6`.

Reviewer: ```suggestion
For models with input dimensions up to `300`, the parameter `n_bits` can be set to `8` or more. When the input dimensions are larger, `n_bits` must be reduced to `6-7`. In many cases, quantized models can preserve all performance metrics compared to the non-quantized float models from scikit-learn when `n_bits` is down to `6`. Users should validate accuracy on held-out test sets and adjust `n_bits` accordingly.
```

Author: We can refer to users as "you" => "You should validate accuracy on held-out test sets and adjust `n_bits` accordingly"

Reviewer: yes, sure

> The same quantization parameters such as scale and zero-point are applied on all features, so you should try to make all feature distribution similar by using standard or min-max normalization. For a more detailed comparison of the impact of such pre-processing please refer to [the logistic regression notebook](../advanced_examples/LogisticRegression.ipynb).

Reviewer: ```suggestion
{% hint style="warning" %}
For best results the distribution of individual features should be similar. This can be achieved using standardization or by using standard or min-max normalization. When many one-hot features are present, [Principal Component Analysis](https://scikit-learn.org/stable/modules/generated/sklearn.decomposition.PCA.html) can be considered as a pre-processing stage. For a more detailed comparison of the impact of such pre-processing please refer to [the logistic regression notebook](../advanced_examples/LogisticRegression.ipynb).
{% endhint %}
```

Author: {% hint style="warning" %}
For optimal results, you can use standard or min-max normalization to achieve a similar distribution of individual features. When there are many one-hot features, consider [Principal Component Analysis](https://scikit-learn.org/stable/modules/generated/sklearn.decomposition.PCA.html)  as a pre-processing stage. 

For a more detailed comparison of the impact of such pre-processing,  please refer to [the logistic regression notebook](../advanced_examples/LogisticRegression.ipynb).
{% endhint %}

## Pre-trained models

> You can convert an already trained scikit-learn linear model to a *Concrete ML one by using the [`from_sklearn_model`](../references/api/concrete.ml.sklearn.base.md#classmethod-from_sklearn_model) method. See [the following example](linear.md#loading-a-pre-trained-model). This functionality is only available for linear models.

Reviewer: ```suggestion
You can convert an already trained scikit-learn linear model to a *Concrete ML one by using the [`from_sklearn_model`](../references/api/concrete.ml.sklearn.base.md#classmethod-from_sklearn_model) method. See [the following example](linear.md#loading-a-pre-trained-model). 
```

# Output:
#  100 examples over 100 have an FHE inference equal to the clear inference
```
## Model accuracy

> We can plot the decision boundary of the classifier and compare those results with a scikit-learn model executed in clear. You can find the complete code in the [LogisticRegression notebook](../tutorials/ml_examples.md).

Reviewer: ```suggestion
The figure below shows how to plot the decision boundary of the FHE classifier and compares those results with a scikit-learn model executed in clear. You can find the complete code in the [LogisticRegression notebook](../tutorials/ml_examples.md).
```

Author: The figure below compares the decision boundary of the FHE classifier and a scikit-learn model executed in clear.

> The overall accuracy scores are identical (93%) between the scikit-learn model (executed in the clear) and the *Concrete ML one (executed in FHE). In fact, quantization has little impact on the decision boundaries, as linear models can use large precision numbers when quantizing inputs and weights in *Concrete ML. Additionally, as the linear models do not use [Programmable Boostrapping](/getting-started/concepts.md#cryptography-concepts), the FHE computations are always exact. This ensures that the FHE predictions are always identical to the quantized clear ones.

Reviewer: ```suggestion
The overall accuracy scores are identical (93%) between the scikit-learn model (executed in the clear) and the *Concrete ML one (executed in FHE). In fact, quantization has little impact on the decision boundaries, as linear models can use large precision numbers when quantizing inputs and weights in *Concrete ML. Additionally, as the linear models do not use [Programmable Boostrapping](/getting-started/concepts.md#cryptography-concepts), the FHE computations are always exact, irrespective of the [PBS error tolerance setting](../explanations/advanced_features.md#approximate-computations). This ensures that the FHE predictions are always identical to the quantized clear ones.
```
